modular_skyrat/master_files/code/modules/mob/living/carbon/human/species_type/snail.dm

```diff
@@ -1,6 +1,42 @@
+#define SHELL_TRANSPARENCY_ALPHA 90
+
+/obj/item/storage/backpack/snail
+	var/storage_core = FALSE
+
 /obj/item/storage/backpack/snail/Initialize(mapload)
 	. = ..()
-	atom_storage.max_specific_storage = 30
+	atom_storage.max_total_storage = 30
+
+/obj/item/storage/backpack/snail/attackby(obj/item/core, mob/user)
+	if(istype(core, /obj/item/assembly/signaler/anomaly/bluespace))
+		to_chat(user, span_notice("You insert [core] into your shell, and it starts to glow blue with expanded storage potential!"))
+		playsound(src.loc, 'sound/machines/click.ogg', 50, TRUE)
+		add_filter("bluespace_shell", 2, list("type" = "outline", "color" = COLOR_BLUE_LIGHT, "size" = 1))
+		storage_core = TRUE
+		qdel(core)
+		emptyStorage()
+		create_storage(max_specific_storage = WEIGHT_CLASS_GIGANTIC, max_total_storage = 35, max_slots = 30, storage_type = /datum/storage/bag_of_holding)
+		atom_storage.allow_big_nesting = TRUE
+		name = "snail shell of holding"
+		user.update_worn_back()
+		update_appearance()
+		return
+	return ..()
```

Comment:
This is a case where you'd want to have it be an early return, send me a message on discord if you don't know what I mean by that.